src/validator/rules/FragmentsOnCompositeTypes.js

```diff
@@ -29,21 +30,22 @@ export default function FragmentsOnCompositeType(
 ): any {
   return {
     InlineFragment(node) {
-      var typeName = node.typeCondition.value;
-      var type = context.getSchema().getType(typeName);
-      if (!isCompositeType(type)) {
+      var type = context.getType();
+      if (type && !isCompositeType(type)) {
         return new GraphQLError(
-          `Fragment cannot condition on non composite type "${typeName}".`,
+          inlineFragmentOnNonCompositeErrorMessage(print(node.typeCondition)),
           [node.typeCondition]
         );
       }
     },
     FragmentDefinition(node) {
-      var typeName = node.typeCondition.value;
-      var type = context.getSchema().getType(typeName);
-      if (!isCompositeType(type)) {
+      var type = context.getType();
+      if (type && !isCompositeType(type)) {
         return new GraphQLError(
-          fragmentOnNonCompositeErrorMessage(node.name.value, typeName),
+          fragmentOnNonCompositeErrorMessage(
+            node.name.value,
+            print(node.typeCondition)
+          ),
```

Comment:
Related. `typeCondition` changed from `Name` to `NamedType` which necessitated changing how it's printed.